### DISCUSSION TITLE
ChatGPT Plugins

### DISCUSSION THREAD

**Discussion by lukemt** [bramses/chatgpt-md#48] [Ideas]
Hi guys!

What are your ideas on integrating ChatGPT Plugins?

I know there is no official way to implement this yet, but the OpenAI guys have already written about this [here](https://openai.com/blog/chatgpt-plugins): 
> We’re starting with a small set of users and are planning to gradually roll out larger-scale access as we learn more (for plugin developers, ChatGPT users, and after an alpha period, _**API users who would like to integrate plugins** into their products_)

It seems there will be something official in the future, which is pretty exciting.

Besides from this at least in theory there are some other unofficial ways one could implement this kind of functionality.
For example I found the following repo here on github (haven't tested yet): https://github.com/eg9y/chat-api-plugins

And then there are ideas for new plugins like:
- Search and retrieve notes, reason about existing notes ect.
- A Tasklist plugin that can view edit and reason about a markdown task list file for example
- A reminder plugin that can set reminders 
- Plus: Use all the existing and soon to be existing chatgpt plugins out there

This is an open discussion, only exchanging ideas at this moment

**Comment by bramses** [comment]
i think this sounds awesome, but id rather be patient and wait till an official endpoint is out before i add plugins. i had a similar thought process before the chatgpt api came out but decided the hoops werent worth the effort

> Search and retrieve notes, reason about existing notes ect.

check out https://github.com/openai/chatgpt-retrieval-plugin if you havent!

> A Tasklist plugin that can view edit and reason about a markdown task list file for example

funnily enough, im building a chatgpt plugin about this rn: https://www.bramadams.dev/202304052333/ i'll post here how it goes!
